Sming/SmingCore/Network/TcpClient.cpp

```diff
@@ -53,6 +53,14 @@ TcpClient::~TcpClient()
 	}
 }
 
+bool TcpClient::connect(const char * server, int port, boolean useSsl /* = false */, uint32_t sslOptions /* = 0 */)
+{
+	if (isProcessing()) return false;
+
+	state = eTCS_Connecting;
+	return TcpConnection::connect(server, port, useSsl, sslOptions);
```

Comment:
I changed TcpClient code to call cstr version from String version of connect() and left firmware running for couple of hours calling simple tcp echo server every 10 seconds, no instabilities observed:

```
diff --git a/Sming/SmingCore/Network/TcpClient.cpp b/Sming/SmingCore/Network/TcpClient.cpp
index 638373f..1d7a2eb 100644
--- a/Sming/SmingCore/Network/TcpClient.cpp
+++ b/Sming/SmingCore/Network/TcpClient.cpp
@@ -63,10 +63,7 @@ bool TcpClient::connect(const char * server, int port, boolean useSsl /* = false
 
 bool TcpClient::connect(String server, int port, boolean useSsl /* = false */, uint32_t sslOptions /* = 0 */)
 {
-       if (isProcessing()) return false;
-
-       state = eTCS_Connecting;
-       return TcpConnection::connect(server.c_str(), port, useSsl, sslOptions);
+       return TcpClient::connect(server.c_str(), port, useSsl, sslOptions);
 }
 
 bool TcpClient::connect(IPAddress addr, uint16_t port, boolean useSsl /* = false */, uint32_t sslOptions /* = 0 */)
```